boskos/janitor/janitor.go

```diff
@@ -0,0 +1,74 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"fmt"
+	"os/exec"
+	"time"
+
+	"github.com/Sirupsen/logrus"
+	"k8s.io/test-infra/boskos/client"
+)
+
+var (
+	clean = janitor_clean
+)
+
+func main() {
+	logrus.SetFormatter(&logrus.JSONFormatter{})
+	boskos := client.NewClient("Janitor", "http://boskos")
+	logrus.Infof("[Janitor]Initialized boskos client!")
+	for range time.Tick(time.Minute * 10) {
+		update(boskos)
+	}
+}
+
+// Clean by janitor script
+func janitor_clean(proj string) error {
+	script := "../../jenkins/janitor.py"
+	return exec.Command(fmt.Sprintf("%s --project=%s --hour=0", script, proj)).Run()
+}
+
+type boskosClient interface {
+	Acquire(rtype string, state string, dest string) (string, error)
+	UpdateAll(state string) error
+	ReleaseOne(name string, dest string) error
+}
+
+func update(c boskosClient) {
+	// Update all holding resources in cleaning state
+	if err := c.UpdateAll("cleaning"); err != nil {
+		logrus.WithError(err).Error("UpdateAll error")
+	}
+
+	// Try to acquire all dirty projects, until none are available.
+	for {
+		if proj, err := c.Acquire("project", "dirty", "cleaning"); err != nil {
+			logrus.WithError(err).Error("Acquire failed!")
```

Comment:
This seems like the expected state of affairs. Can we distinguish between errors and "nothing else is dirty?"